singularity-opac/base/indexers.hpp

```diff
@@ -31,6 +31,19 @@
 namespace singularity {
 namespace indexers {
 
+template <typename T>
+struct Identity {
+  Identity() = default;
+  Identity(T &t) : data_(std::forward<T>(t)) {}
```

Comment:
Is this a copy or move constructor? You are passing an l-value reference, but forwarding it (which is a no-op on l-values). You can use `Identity(T&& t) : data_(std::forward<T>(t)) {}` to let `forward` choose copy or move, but this isn't idiomatic.

Reply:
Oops good catch.